rust/main/utils/run-locally/src/sealevel/mod.rs

```diff
@@ -0,0 +1,328 @@
+pub mod solana;
+pub mod termination_invariant;
+
+use std::{
+    fs,
+    path::Path,
+    sync::atomic::Ordering,
+    thread::sleep,
+    time::{Duration, Instant},
+};
+
+use tempfile::tempdir;
+
+use crate::SHUTDOWN;
+use crate::{
+    config::Config,
+    invariants::post_startup_invariants,
+    logging::log,
+    long_running_processes_exited_check,
+    metrics::agent_balance_sum,
+    program::Program,
+    sealevel::{solana::*, termination_invariant::*},
+    utils::{
+        concat_path, get_sealevel_path, get_ts_infra_path, get_workspace_path, make_static,
+        TaskHandle,
+    },
+    wait_for_condition, State, AGENT_LOGGING_DIR, RELAYER_METRICS_PORT, SCRAPER_METRICS_PORT,
+};
+
+// This number should be even, so the messages can be split into two equal halves
+// sent before and after the relayer spins up, to avoid rounding errors.
+pub const SOL_MESSAGES_EXPECTED: u32 = 20;
+pub const SOL_MESSAGES_WITH_NON_MATCHING_IGP: u32 = 1;
+
+/// These private keys are from hardhat/anvil's testing accounts.
```

Comment:
this comment is a bit stale, should say
```suggestion
/// These private keys are from the solana-test-validator network accounts.
```